plugin/src/main/java/com/h3xstream/findsecbugs/crypto/InsufficientKeySizeRsaDetector.java

```diff
@@ -118,22 +116,21 @@ private void analyzeMethod(Method m, ClassContext classContext) throws CFGBuilde
                         }
                     }
 
-                    if (n != null && n.intValue() < 1024) {
-                        initializeWeakKeyLength = true;
-                        locationWeakness = location;
+                    if (n != null && n.intValue() < 2048 && createRsaKeyGen) {
```

Comment:
Good adjustment. Reporting the issue for 1024 bit key is perfect.
Even NIST which was one of the rare to allow 1024 bit for legacy key. This support has expired for key used in 2016+ (https://www.keylength.com/en/4/)